android/src/main/java/com/vydia/RNUploader/UploadUtils.kt

```diff
@@ -0,0 +1,78 @@
+package com.vydia.RNUploader
+
+import kotlinx.coroutines.suspendCancellableCoroutine
+import okhttp3.*
+import okhttp3.Headers.Companion.toHeaders
+import okhttp3.RequestBody.Companion.asRequestBody
+import okio.Buffer
+import okio.BufferedSink
+import okio.ForwardingSink
+import okio.buffer
+import java.io.File
+import java.io.IOException
+import kotlin.coroutines.resumeWithException
+
+// Throttling interval of progress reports
+private const val PROGRESS_INTERVAL = 500 // milliseconds
+
+
+// make an upload request using okhttp
+suspend fun okhttpUpload(
+  client: OkHttpClient,
+  upload: Upload,
+  file: File,
+  onProgress: (Long) -> Unit
+) =
+  suspendCancellableCoroutine<Response> { continuation ->
```

Comment:
similar to `okhttpUpload = (....) => new Promise((resolve, reject) => ....)` in JS